Add documentation for `snap_getEntropy` (#590)
MetaMask/metamask-docs: docs/guide/snaps-rpc-api.md

::::

### `snap_getEntropy`

::: warning Only Callable By

- Snaps
  :::

#### Parameters

- `Object`

>   - `version` - The literal number `1`. This is to allow for future versions of this method to be added.

Reviewer: ```suggestion
  - `version` - The literal number `1`. This is reserved for future use.
```

>   - `salt` - An arbitrary string to be used as a salt for the entropy. This can be used to generate different entropy

Reviewer: Should we add some more explicit note here about the determinism of the function? While it is described (nicely, I think) in the description, improper salt setting is a common cause of compromise and could warrant a notice for those who average library users who might not grok the implications?

Author: > improper salt setting and handling are common causes of compromise and could warrant a notice for those who might not grok the implications?

The entropy is Snap-specific. If two Snaps use the same salt, the generated entropy is still different. It shouldn't matter what salt is used.

The implementation is described here: https://github.com/MetaMask/SIPs/blob/60c212d1da607005bef989fd364565f21c66f111/SIPS/sip-6.md

    for different purposes.

#### Returns

The entropy as hexadecimal `string`.

#### Description

Gets a deterministic 256-bit entropy value, which is specific to the Snap and the user's account. This entropy can be
used to generate a private key, or any other value that requires a high level of randomness. Other Snaps will not be
able to access this entropy, and it will change if the user's secret recovery phrase changes.

This value is deterministic, meaning that it will always be the same for the same Snap, user account, and salt.

You can optionally specify a salt to generate different entropy for different purposes. Using a salt will result in
completely different entropy, that is unrelated to the entropy generated without a salt.

#### Example

:::: tabs :options="{ useUrlFragment: false }"

::: tab Manifest

```json
{
  "initialPermissions": {
    "snap_getEntropy": {}
  }
}
```

:::

::: tab Code

```javascript

> const entropy = await wallet.request({

Reviewer 2: ```suggestion
const entropy = await snap.request({
```
